src/openfermion/measurements/_hamiltonian_stabilizers_sets.py

```diff
@@ -0,0 +1,126 @@
+    The missing strings overlap with all stabilizers, and will need
+    to be measured even when an error occured.
+
+    Args:
+        Hamiltonian(QubitOperator): Hamiltonian to be split.
+        subsets(list): List of QubitOperators in which the Hamiltonian
+            has been split into.
+    Return:
+        left_paulis(QubitOperator): Only if there are left strings.
+            If no missing Paulis, returns an empyt QubitOperator.
+    """
+    # This functions uses python object sets to compare missing tuples.
+    # Hamiltonian keys as a set.
+    ham_set_form = set(hamiltonian.terms.keys())
+
+    # Hamiltonian subsets as a single set object.
+    aux_set = set()
+    for sb in subsets:
+        aux_set.update(set(sb.terms.keys()))
+
+    # Find any missing tuple in the Hamiltonian set compared to
+    # all subsets.
+    left_set = ham_set_form - aux_set
+
+    # Save missing Pauli strings in QubitOperator
+    # If no missing Paulis, returns 0
+    left_paulis = QubitOperator()
+    for p in left_set:
+        left_paulis += QubitOperator(p, hamiltonian.terms[p])
+    return left_paulis
+
+
+def get_hamiltonian_subsets(hamiltonian, stabilizers):
+    """
+    Create subsets of Hamiltonian.
+
+    This function splits a Hamiltonian into subsets of Pauli strings that do
+    not overlap with stabilizers.
+    Pauli strings might be repeated between subsets if they do not share any
+    qubit with the stabilizer.
+    The Pauli strings that overlap with all stabilizers will be stored with a
+    separate set.
+    It can be used to perform SV and not disregarding all data if only a single
+    stabilizers signals an error.
+
+    Args:
+        hamiltonian(QubitOperator): Hamiltonian to be split.
+        stabilizers(list or QubitOperator): List of stabilizers as
+            QubitOperators.
+    Return:
+        hamiltonian_subsets(list): List of QubitOperator which are a subset of
+            the Hamiltonian.
+            Each QubitOperator does not overlap with the stabilizer at the same
+            position in the list.
+        rest_paulis(QubitOperator): Operator that contains the Pauli
+            strings that overlap with all the stabilizers.
+            If there are no paulis left an empty QubitOperator is returned.
+    Raises:
+        TypeError: Input hamiltonian must be QubitOperator.
+        TypeError: Input stabilizer_list must be QubitOperator or list.
+    """
+    if not isinstance(hamiltonian, QubitOperator):
+        raise TypeError('Input terms must be QubitOperator.')
+    if not isinstance(stabilizers, (QubitOperator, list,
+                                    tuple, numpy.ndarray)):
+        raise TypeError('Input stabilizers must be QubitOperator or list.')
+
+    hamiltonian_subsets = []
+    for stab in list(stabilizers):
+        # Initialize subset
+        ham_subset = QubitOperator()
+        for pauli in hamiltonian:
+            if _has_overlapping_indices(pauli, stab) is False:
```

Comment:
It's usually clearer to have a positive conditional rather than a negative one. Change the function `_has_overlapping_indices` to `_has_disjoint_indices` (which returns the opposite value) and use that instead. Of course, update the documentation of that function too.

Reply:
Changed to a positive conditional.